viewer/event_db.cpp

```diff
@@ -44,7 +55,24 @@ std::vector<EventRef> EventDB::get_data(HT_TimestampNs start_ts,
     }    
     else
     {
-        _query_event_klass(response, _events[query.klass_id], start_ts, stop_ts); 
+        if (_events.find(query.klass_id) == _events.end())
+            return response;
+
+        std::vector<parser::Event>::iterator first_event;
+        std::vector<parser::Event>::iterator last_event;
+        _get_range_of_events(_events[query.klass_id], start_ts, stop_ts, first_event, last_event);
+        --last_event;
+
+        if (_cache.range_in_cache(first_event->get_timestamp(), last_event->get_timestamp(), query.klass_id))
```

Comment:
you could use `std::prev(last_event)->get_timestamp()` so you don't have to modify `last_event` object twice